src/main/kotlin/no/nav/amt_enhetsregister/service/DeltaOppdateringEnhetService.kt

```diff
@@ -59,31 +59,34 @@ class DeltaOppdateringEnhetService(
 
 			if (underenhet == null) { // GONE
 				log.info("Underenhet orgnr=${it.organisasjonsnummer} er fjernet fra brreg")
+				val opprinneligOverordnetEnhet = enhetService.hentEnhet(it.organisasjonsnummer)?.overordnetEnhetOrganisasjonsnummer
 
 				return@deltaOppdaterEnhet EnhetService.UpsertEnhet(
 					organisasjonsnummer = it.organisasjonsnummer,
 					navn = "Fjernet virksomhet",
-					overordnetEnhetOrgNr = UKJENT_VIRKSOMHET_NR
+					overordnetEnhetOrgNr = opprinneligOverordnetEnhet
```

Comment:
Jeg tror et problem med dette er at, hvis `overordnetEnhetOrgNr` er `null` så mister vi informasjonen om at dette er en underenhet. Det har kanskje ikke så mye å si for oss.

Hvis vi uansett har lagret enheten kan vel navnet beholdes også med et `(slettet)` tillegg slik som nedenfor.

Reply:
Ja, jeg lurte litt på det med om vi skulle ha beholdt navnet. Vi kan kanskje like godt gjøre det? 

Hvis overordnetEnhetOrgNr er null her så betyr vel det at den var null før man startet oppdateringen også? Og da burde den kanskje ikke blitt tolket som en underenhet i utgangspunktet? Eller misforstår jeg nå? 😅

Reply:
brrg sitt endepunkt er spesifikt for underenheter.

Reply:
Ja, ser det :) Synes koden er veldig kronglete å lese, sliter med å se hvordan den finner ut om noe er moderenhet eller underenhet, men det må vel være basert på om vi har lagret overordnet enhet for den, eller?

Reply:
Den henter vel bare de siste oppdateringene fra brreg tror jeg:
```		val oppdateringer =
			if (enhetType == EnhetType.MODERENHET)
				bronnoysundClient.hentModerenhetOppdateringer(progresjon.oppdateringId, OPPDATERINGER_SIZE)
			else
				bronnoysundClient.hentUnderenhetOppdateringer(progresjon.oppdateringId, OPPDATERINGER_SIZE)
```

Reply:
åja, for den har bare et bokmerke den leser fra, på en måte? Men hvis vi får en oppdatering på underenhet-endepunktet så har jo enheten vært en underenhet? Og hvis vi ikke har lagret noen overordnet enhet tidligere for den, så mister vi vel egentlig ingenting? 

Det er jo tillatt at enheter ikke har en overordnet enhet, og tilganger er uansett knyttet til underenheten. I aktivitetsplanen viser vi navn fra overordnet enhet hvis den finnes, og eller bruker vi navn på underenhet. Da klarer jeg ikke helt å se at noe blir feil om vi skulle komme til å sette overordnet enhet til null hvis vi av en eller annen grunn mangler den?

Reply:
Jeg kan evt sjekke at den overordnede enheten vi har lagret ikke er null, og hvis den er null så bruker vi default-orgnummer i stedet? Selv om jeg synes det må være bedre å sette null enn å sette en default-verdi. Vi er vel interessert i navn/nummer på organisasjonene, ikke brreg sin organisering av over- og underenheter? 😅

Reply:
Det er nok bare en detalj for oss egentlig, men sånn logisk sett så kan man ikke ha en underenhet uten en hovedenhet:

> En underenhet identifiserer den aktiviteten hovedenheten driver. En hovedenhet kan ha en eller flere  underenheter. Underenhetene er ikke selvstendige. De vil alltid være knyttet opp mot en hovedenhet. Underenhet er virksomhet på laveste nivå i registreringsstrukturen i Enhetsregisteret.
https://www.brreg.no/bedrift/underenhet/

For oss så har nok ikke dette noe å si, men jeg vet ikke med andre som konsumerer den.

Reply:
Vi har tatt diskusjonene på discord og slack og blitt enige om at det ikke er så viktig hverken for oss eller DAB å beholde informajson om hvorvidt en enhet har vært en underenhet eller ikke :)